src/main/java/se/sundsvall/parkingpermit/businesslogic/worker/execution/HandleLostCardTaskWorker.java

```diff
@@ -0,0 +1,101 @@
+import org.camunda.bpm.client.task.ExternalTask;
+import org.camunda.bpm.client.task.ExternalTaskService;
+import org.springframework.stereotype.Component;
+import se.sundsvall.parkingpermit.businesslogic.handler.FailureHandler;
+import se.sundsvall.parkingpermit.businesslogic.worker.AbstractTaskWorker;
+import se.sundsvall.parkingpermit.integration.camunda.CamundaClient;
+import se.sundsvall.parkingpermit.integration.casedata.CaseDataClient;
+import se.sundsvall.parkingpermit.service.PartyAssetsService;
+
+@Component
+@ExternalTaskSubscription("HandleLostCardTask")
+public class HandleLostCardTaskWorker extends AbstractTaskWorker {
+
+	private static final String NOTE_TEXT = "The asset with id %s has been blocked.";
+	private static final String NOTE_TITLE = "Asset blocked";
+	private static final String PARTY_ASSET_STATUS_REASON = "LOST";
+
+	private final PartyAssetsService partyAssetsService;
+
+	HandleLostCardTaskWorker(CamundaClient camundaClient, CaseDataClient caseDataClient, FailureHandler failureHandler, PartyAssetsService partyAssetsService) {
+		super(camundaClient, caseDataClient, failureHandler);
+		this.partyAssetsService = partyAssetsService;
+	}
+
+	@Override
+	public void executeBusinessLogic(ExternalTask externalTask, ExternalTaskService externalTaskService) {
+		try {
+			final String municipalityId = getMunicipalityId(externalTask);
+			final String namespace = getNamespace(externalTask);
+			final Long caseNumber = getCaseNumber(externalTask);
+
+			final var errand = getErrand(municipalityId, namespace, caseNumber);
+
+			if (CASE_TYPE_LOST_PARKING_PERMIT.equals(errand.getCaseType())) {
+				final var assets = partyAssetsService.getAssets(municipalityId, getStakeholderPersonIdOfApplicant(errand), PARTY_ASSET_STATUS_ACTIVE);
+
+				final var existingAsset = Optional.ofNullable(assets).orElse(emptyList()).stream()
+					.filter(asset -> PARTY_ASSET_TYPE.equals(asset.getType()))
+					.findFirst();
+
+				existingAsset.ifPresent(asset -> {
+					partyAssetsService.updateAssetWithNewStatus(municipalityId,
+						asset.getId(), BLOCKED, PARTY_ASSET_STATUS_REASON);
+
+					final var extraParameters = Optional.ofNullable(errand.getExtraParameters()).orElse(new ArrayList<>());
+
+					if (extraParameters.stream().noneMatch(extraParameter -> CASEDATA_KEY_ARTEFACT_LOST_PERMIT_NUMBER.equals(extraParameter.getKey()))) {
+						extraParameters.add(new ExtraParameter().key(CASEDATA_KEY_ARTEFACT_LOST_PERMIT_NUMBER).values(List.of(asset.getAssetId())));
+					} else {
+						extraParameters.stream()
+							.filter(extraParameter -> CASEDATA_KEY_ARTEFACT_LOST_PERMIT_NUMBER.equals(extraParameter.getKey()))
+							.findFirst()
+							.ifPresent(extraParameter -> extraParameter.setValues(List.of(asset.getAssetId())));
+					}
+
+					caseDataClient.patchErrand(municipalityId, namespace, errand.getId(), new PatchErrand().extraParameters(extraParameters).facilities(null));
```

Comment:
I'm just curious:
Why do you need to set facilities to null here?

Reply:
I don't know the reason but we did the same in mapper so I thougth it was for a reason. I have changed in both places and we send an empty list now.